cachito/web/auth.py

```diff
@@ -0,0 +1,65 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import logging
+
+from flask import current_app
+
+from cachito.web import db
+from cachito.web.models import User
+
+
+log = logging.getLogger(__name__)
+
+
+def user_loader(username):
+    """
+    Get the user by their username from the database.
+
+    This is used by the Flask-Login library.
+
+    :param str username: the username of the user
+    :return: the User object associated with the username or None
+    :rtype: cachito.web.models.User
+    """
+    return User.query.filter_by(username=username).first()
+
+
+def load_user_from_request(request):
+    """
+    Load the user that authenticated from the current request.
+
+    When authentication is turned on (default in production), this relies on the "REMOTE_USER"
+    environment variable being set. This is usually set by the mod_auth_gssapi Apache authentication
+    module.
+
+    This is used by the Flask-Login library. If None is returned, then Flask-Login will set
+    `flask_login.current_user` to an `AnonymousUserMixin` object, which has the `is_authenticated`
+    property set to `False`. Additionally, any route decorated with `@login_required` will raise
+    an `Unauthorized` exception.
+
+    :param flask.Request request: the Flask request
+    :return: the User object associated with the username or None
+    :rtype: cachito.web.models.User
+    """
+    remote_user = request.environ.get('REMOTE_USER')
+    if not remote_user:
+        if current_app.config.get('LOGIN_DISABLED', False) is True:
+            log.info(
+                'The REMOTE_USER environment variable wasn\'t set on the request, but the '
+                'LOGIN_DISABLED configuration is set to True.'
+            )
+        return
+
+    log.info(f'The user "{remote_user}" was authenticated successfully by httpd')
+    if '@' in remote_user:
```

Comment:
I'm not sure. I suppose it was just because we have one Kerberos realm, and it seemed cleaner to just store the username, but you're right. We might as well keep it. I'll change it to all lower-case though in case capitalization comes out differently. Freshmaker currently does this as well, so I don't know if having it consistent matters to you.